man/tpm2_sign.1.md

```diff
@@ -2,20 +2,29 @@
 
 # NAME
 
-**tpm2_sign**(1) - Sign a hash using the TPM.
+**tpm2_sign**(1) - Sign a hash or message using the TPM.
 
 # SYNOPSIS
 
 **tpm2_sign** [*OPTIONS*] [*ARGUMENT*]
 
 # DESCRIPTION
 
-**tpm2_sign**(1) - Signs an externally provided message or hash with the
-specified symmetric or asymmetric signing key.
+**tpm2_sign**(1) - Generates signature of specified message or message-digest
+using the specified symmetric or asymmetric signing key.
 
-If the signing key is a restricted signing key, then validation can be provided
-via the **-t** output. The ticket indicates that the TPM performed the hash of
-the message.
+When signing a message, **tpm2_sign** utility first calculates the digest of the
+message like in the **tpm2_hash**. It also generates a validation ticket under
+TPM2_RH_NULL or TPM2_RH_OWNER hierarchies respectively for un-restricted or the
+restricted signing keys.
+
+While signing messages is a provision in this tool it is recommended to use the
+**tpm2_hash** tool first and pass the digest and validation ticket for
+flexibility in choice of message length and hierarchy of choice for generating
```

Comment:
How does it affect message length?

Reply:
Ah... i was looking at the lib/tpm2_hash.c where message lengths longer than max buffer were split and hashed in loops. I will drop that comment.

Reply:
> Ah... i was looking at the lib/tpm2_hash.c where message lengths longer than max buffer were split and hashed in loops. I will drop that comment.

Im confused, if you were looking at that code, wouldn't that mean that message lengths of any size are supported?